### PR TITLE
Update metadata in setup.py

### LINKED ISSUE #595: update the long description in setup.py

### PR DESCRIPTION
**Pull request by Carreau**
closes #595

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -42,16 +42,44 @@
     exec(f.read(), {}, version_ns)
 
 
+long_desc = """
```
Comment (rmorshea) [inline setup.py:45]: Can we just read in the `README.md` and use that as the `long_desc`?

```python
with open(os.path.join(here, "README.md")) as f:
    long_description = f.read()

long_description = long_description
long_description_content_type = "text/markdown"
```

Comment (Carreau) [reply setup.py:45]: We could, but there are some github specific thing in the readme. Not sure it matters much.

File: setup.py [hunk 2]
```diff
@@ -42,16 +42,44 @@
...
+
+Its implementation relies on the [descriptor](https://docs.python.org/howto/descriptor.html)
+pattern, and it is a lightweight pure-python alternative of the
+[*traits* library](http://code.enthought.com/pages/traits.html).
+
+Traitlets powers the configuration system of IPython and Jupyter
+and the declarative API of IPython interactive widgets.
+"""
+
+
```
Comment (Carreau) [inline setup.py:66]:
```suggestion
with open(os.path.join(here, "README.md")) as f:
    long_description = f.read()

long_description_content_type = "text/markdown"
```

File: setup.py [hunk 3]
```diff
@@ -42,16 +42,44 @@
...
 setup_args = dict(
     name            = name,
     version         = version_ns['__version__'],
     scripts         = glob(pjoin('scripts', '*')),
     packages        = packages,
-    description     = "Traitlets Python config system",
-    long_description= "A configuration system for Python applications.",
+    description     = "Traitlets Python configuration system",
+    long_description= long_desc,
```
Comment (Carreau) [inline setup.py:73]:
```suggestion
    long_description= long_description,
```

**Pull request merged**